src/boot/efi/util.c

```diff
@@ -430,3 +430,21 @@ VOID print_at(UINTN x, UINTN y, UINTN attr, const CHAR16 *str) {
         uefi_call_wrapper(ST->ConOut->SetAttribute, 2, ST->ConOut, attr);
         uefi_call_wrapper(ST->ConOut->OutputString, 2, ST->ConOut, (CHAR16 *)str);
 }
+
+VOID draw_box(UINTN x, UINTN y, UINTN w, UINTN h, UINTN color) {
+        print_at(x, y, color, L"┌");
+        print_at(x + w, y, color, L"┐");
+        print_at(x, y + h, color, L"└");
+        print_at(x + w, y + h, color, L"┘");
```

Comment:
hmm, so far we refrained from non-ascii chars in our own choice of characters in sd-boot. I doubt this is safe to do on plenty firmware, and definitely not when used on serial terminal.

I am not convinced this is a good idea. Note that the original design originates as a gnome design group design, this is just an implementaiton of it. I am not convinced the box is really someting we want, what purpose does it serve really?

Reply:
The efi spec requires these box drawing characters to be supported. If you want, I can do a safety check and fall back to ascii.

Does sd-boot even work over serial terminal? How can I test this with OVMF/qemu?

Reply:
how would you do a safety check?

iirc it suffices to just tell qemu to boot with a serial terminal instead of a console. tianocore figures that out on its own then, and it just works

Reply:
There's a TestString function that is supposed to tell you if the string can be shown.

Reply:
smells like one of those apis that are likely lying on many firmwares?

Reply:
Hence, why I didn't use it. Imho, these box characters should be fine since they are mandated (they even have constants for it in the headers). MokManager is using them, btw.

Reply:
mokmanager has no fallback?

i am mostly concerned about serial port access, i am pretty sure efi will only pass ascii over there, but dunno

Reply:
From what it looks like, they do it unconditionally.

I just tried with -serial stdio and output mostly works, input still works, but you get garbage for the box chars:
![image](https://user-images.githubusercontent.com/7892635/127687880-0732a9a8-689d-4fc0-8d07-35dea3cf38ac.png)

I would expect the TestString still reports success then. I still like the look of the box though...

I can make it not print them if box-color foreground==background (of color)?

Reply:
not sure i follow? 

as you noticed the box looks crap on serial ports, i'd hence really just not do it. I don't rok what the box should bring as a benefit

Reply:
It really does look nice imho (okay, subjective, but I've grown to like it very much the last few days of using it). And I've made it a config option (default off) so we don't get grumpy serial port user complaints.